File: regexlib/PoCs/Regexlib_Java_8/src/regexlib_8181.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class regexlib_8181 {
//    /* 8181
//     * <script.*/*>|</script>|<[a-zA-Z][^>]*=['"]+javascript:\w+.*['"]+>|<\w+[^>]*\son\w+=.*[ /]*>
//     * POLYNOMIAL
//     * nums:2
//     * POLYNOMIAL AttackString:""+"<script"*5000+"! _1SLQ_2"
//     */
    public static void main(String[] args) throws InterruptedException {
        String regex = "<script.*/*>|</script>|<[a-zA-Z][^>]*=[\'\"]+javascript:\\w+.*[\'\"]+>|<\\w+[^>]*\\son\\w+=.*[ /]*>";
        for (int i = 0; i < 1000; i++) {
            StringBuilder attackString = new StringBuilder();
            // 前缀
            attackString.append("");
            // 歧义点
            for (int j = 0; j < i * 10000; j++) {
                attackString.append("<script");
            }
            // 后缀
            attackString.append("! _1SLQ_2");
//            System.out.println(attackString);
            long time1 = System.nanoTime();
//            boolean isMatch = Pattern.matches(regex, attackString);
            boolean isMatch = Pattern.compile(regex).matcher(attackString).find();
            long time2 = System.nanoTime();
            System.out.println(i * 10000 + " " + isMatch + " " + (time2 - time1)/1e9);
        }
    }
}

File: regexlib/python_re2_test_file/regexlib_8181.py
# 8181
# <script.*/*>|</script>|<[a-zA-Z][^>]*=['"]+javascript:\w+.*['"]+>|<\w+[^>]*\son\w+=.*[ /]*>
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"<script"*5000+"! _1SLQ_2"

import re2 as re
from time import perf_counter

regex = """<script.*/*>|</script>|<[a-zA-Z][^>]*=['"]+javascript:\w+.*['"]+>|<\w+[^>]*\son\w+=.*[ /]*>"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "<script" * i * 10000 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")